### PR TITLE
Refactor to DelegatorClaim and implement new MsgClaimDelegatorReward

### PR DESCRIPTION
**Pull request by denalimarsh**
READ BEFORE REVIEWING:
1. I haven't tested these changes with kvtool yet or fully tested the querier. I'll be doing more testing over the next few hours.
2. In the migration it's impossible to distinguish between rewards from delegation vs. liquidity providing as they're all combined inside `claim.Reward`. Right now I'm just putting them all inside the Hard claim to avoid duplicating rewards.

### CODE REVIEW

File: x/incentive/keeper/rewards_supply.go [hunk 1]
```diff
@@ -187,8 +187,8 @@ func (k Keeper) SynchronizeHardLiquidityProviderClaim(ctx sdk.Context, owner sdk
 		k.SynchronizeHardBorrowReward(ctx, borrow)
 	}
 
-	// Synchronize any hard delegator rewards
-	k.SynchronizeHardDelegatorRewards(ctx, owner, nil, false)
+	// Synchronize any delegator rewards
```
Comment (karzak) [inline x/incentive/keeper/rewards_supply.go:190]: Why is it necessary to sync delegator rewards here?

**Review by karzak — CHANGES_REQUESTED** [review]
Looks solid, just a few minor things and clarifications

File: x/incentive/types/msg.go [hunk 2]
```diff
@@ -184,3 +184,90 @@ func (msg MsgClaimHardRewardVVesting) GetSignBytes() []byte {
 func (msg MsgClaimHardRewardVVesting) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{msg.Sender}
 }
+
+// MsgClaimDelegatorReward message type used to claim delegator rewards
+type MsgClaimDelegatorReward struct {
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/incentive/types/msg.go:189]: Good catch. Looks like `DelegatorClaim` isn't currently registered either but the other claim types are. I'll add it as well.

[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/incentive/types/msg.go:189]: Updated in https://github.com/Kava-Labs/kava/pull/948/commits/0aca99ceff184741c960d4ec285c5e3b4d27213f

File: x/incentive/types/msg.go [hunk 3]
```diff
@@ -184,3 +184,90 @@ func (msg MsgClaimHardRewardVVesting) GetSignBytes() []byte {
...
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners returns the addresses of signers that must sign.
+func (msg MsgClaimDelegatorReward) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// MsgClaimDelegatorRewardVVesting message type used to claim delegator rewards for validator vesting accounts
+type MsgClaimDelegatorRewardVVesting struct {
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/incentive/types/msg.go:230]: Updated in https://github.com/Kava-Labs/kava/pull/948/commits/0aca99ceff184741c960d4ec285c5e3b4d27213f

File: x/incentive/keeper/rewards_supply.go (hunk 1 above)
Comment (denalimarsh) [reply x/incentive/keeper/rewards_supply.go]: Good catch. Removed in https://github.com/Kava-Labs/kava/pull/948/commits/16e95aa1bcd8f71fc870cc7684de2e30865620f5.

File: migrate/v0_15/migrate.go [hunk 4]
```diff
@@ -326,8 +327,14 @@ func Incentive(incentiveGS v0_14incentive.GenesisState) v0_15incentive.GenesisSt
 		delegatorMultiRewardIndex := v0_15incentive.NewMultiRewardIndex(v0_15incentive.BondDenom, delegatorRewardIndexes)
 		delegatorMultiRewardIndexes = append(delegatorMultiRewardIndexes, delegatorMultiRewardIndex)
 
+		// TODO: It's impossible to distinguish between rewards from delegation vs. liquidity providing
```
[the remark this answers is not in the document]
Comment (rhuairahrighairidh) [reply migrate/v0_15/migrate.go:330]: Seems ok for now, I'm planning to go over migrations in detail in future PR.

**Review by karzak — APPROVED** [review]
tACK

**Pull request merged**